src/cascadia/inc/cppwinrt_utils.h

```diff
@@ -55,4 +55,32 @@ private:
 // Use this if you have a Windows.Foundation.TypedEventHandler
 #define DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(className, name, eventHandler, sender, args)                                                        \
     winrt::event_token className::name(Windows::Foundation::TypedEventHandler<sender, args> const& handler) { return eventHandler.add(handler); } \
-    void className::name(winrt::event_token const& token) noexcept { eventHandler.remove(token); }
\ No newline at end of file
+    void className::name(winrt::event_token const& token) noexcept { eventHandler.remove(token); }
+
+// This is a helper method for deserializing a SAFEARRAY of
+// COM objects and converting it to a vector that
+// owns the extracted COM objects
+template<typename T>
+std::vector<wil::com_ptr<T>> SafeArrayToOwningVector(SAFEARRAY* safeArray)
+{
+    T** pVals;
+    THROW_IF_FAILED(SafeArrayAccessData(safeArray, (void**)&pVals));
+
+    long lBound, uBound;
+    THROW_IF_FAILED(SafeArrayGetLBound(safeArray, 1, &lBound));

```

Comment:
You should be asserting something about the number of dimensions in the `SAFEARRAY`. If someone decides to use this with a 2+ dimensional array, it probably won't do what they think it should and should probably throw E_UNEXPECTED or E_NOTIMPL.